gridstatus/spp.py

```diff
@@ -611,8 +611,17 @@ def _fs_get_dam_lmp_by_location_paths(self, date, verbose=False):
             type="folder",
             path=date.strftime("/%Y/%m/By_Day"),
         )
-        max_name = max(files_df["name"])
-        max_file = files_df[files_df["name"] == max_name]
+
+        files_df["date"] = files_df.name.apply(
+            lambda x: pd.to_datetime(
+                x.strip(".csv").split("-")[-1],
+                format="%Y%m%d%H%M",
+            )
+            .normalize()
+            .tz_localize(self.default_timezone),
+        )
+
+        max_file = files_df[files_df["date"] == date]
```

Comment:
this looks great!

I would rename `max_file` to be something like `matched_file` since it is no longer the max date, just the file that matches the requested date.

Reply:
Updated. Thanks!